### PR TITLE
Summing compaction fixes

### PR DESCRIPTION
**Pull request by vavrusa**
This fixes several problems in test related to compaction:

* Keys with zero values are compacted in `sumMap`
* Fixed SummingMergeTree in case an explicit list of columns to sum is given in configuration
* Summation of non-UInt64 columns is fixed, the aggregation now uses `sumWithOverflow(x)` which is a variant of `sum(x)` function that uses same data type for return value as the input type

### CODE REVIEW

File: dbms/src/AggregateFunctions/AggregateFunctionSum.cpp [hunk 1]
```diff
@@ -26,6 +39,7 @@ AggregateFunctionPtr createAggregateFunctionSum(const std::string & name, const
 void registerAggregateFunctionSum(AggregateFunctionFactory & factory)
 {
     factory.registerFunction("sum", createAggregateFunctionSum, AggregateFunctionFactory::CaseInsensitive);
+    factory.registerFunction("sumWithOverflow", createAggregateFunctionSumWithOverflow, AggregateFunctionFactory::CaseInsensitive);
```
Comment (alexey-milovidov) [inline dbms/src/AggregateFunctions/AggregateFunctionSum.cpp]: Don't need to be case insensisive.

File: dbms/src/AggregateFunctions/AggregateFunctionSum.cpp [hunk 2]
```diff
@@ -13,6 +13,19 @@ AggregateFunctionPtr createAggregateFunctionSum(const std::string & name, const
...
+
+    return res;
+}
+
+AggregateFunctionPtr createAggregateFunctionSumWithOverflow(const std::string & name, const DataTypes & argument_types, const Array & parameters)
+{
+    if (argument_types.size() != 1)
+        throw Exception("Incorrect number of arguments for aggregate function " + name, ErrorCodes::NUMBER_OF_ARGUMENTS_DOESNT_MATCH);
+
     AggregateFunctionPtr res(createWithNumericType<AggregateFunctionSum>(*argument_types[0]));
```
Comment (alexey-milovidov) [inline dbms/src/AggregateFunctions/AggregateFunctionSum.cpp]: ?

File: dbms/src/AggregateFunctions/AggregateFunctionSum.cpp [hunk 3]
```diff
@@ -13,6 +13,19 @@ AggregateFunctionPtr createAggregateFunctionSum(const std::string & name, const
     if (argument_types.size() != 1)
         throw Exception("Incorrect number of arguments for aggregate function " + name, ErrorCodes::NUMBER_OF_ARGUMENTS_DOESNT_MATCH);
 
+    AggregateFunctionPtr res(createWithNumericTypeNearest<AggregateFunctionSum>(*argument_types[0]));
```
Comment (vavrusa) [inline dbms/src/AggregateFunctions/AggregateFunctionSum.cpp]: This just instantiates a different variant of the template with `AggregateFunctionSum<T, NearestFieldType<T>::Type>`.

**Comment by ludv1x** [comment]
autotests

**Comment by vavrusa** [comment]
@ludv1x can you repost which tests have failed?

**Comment by ludv1x** [comment]
@vavrusa 
stateless.00155_merges
```
Received exception from server:
Code: 9. DB::Exception: Received from localhost:9000, ::1. DB::Exception: Sizes of columns doesn't match: d: 2, s: 0. 
Received exception from server:
Code: 9. DB::Exception: Received from localhost:9000, ::1. DB::Exception: Sizes of columns doesn't match: d: 2, s: 0. 
Received exception from server:
Code: 9. DB::Exception: Received from localhost:9000, ::1. DB::Exception: Sizes of columns doesn't match: d: 2, s: 0. 
```

This test is in public part of the test suite (`queries/0_statetles/00155...`).

**Comment by vavrusa** [comment]
@ludv1x is there a script to run these manually or as part of the build?

**Comment by vavrusa** [comment]
@ludv1x thanks Vitaliy, that was super helpful, I fixed the mistake.

**Comment by ludv1x** [comment]
> is there a script to run these manually or as part of the build?

`./release` the script only builds a deb package, so you should run `./clickhouse-test` manually.

**Comment by vavrusa** [comment]
Brilliant, this is great, thanks @ludv1x

```
$ /usr/local/bin/python2 ../dbms/tests/clickhouse-test -b "$(pwd)/clickhouse-server" -c "$(pwd)/clickhouse-client" --no-zookeeper --no-shard -q $(pwd)/../dbms/tests/queries "(sum|merges)"

Running stateless tests.

00507_sumwithoverflow:                                                 [ OK ]
00502_sum_map:                                                         [ OK ]
00327_summing_composite_nested:                                        [ OK ]
00322_disable_checksumming:                                            [ OK ]
00155_merges:                                                          [ OK ]
00148_summing_merge_tree_nested_map_multiple_values:                   [ OK ]
00146_summing_merge_tree_nested_map:                                   [ OK ]
00084_summing_merge_tree:                                              [ OK ]
00043_summing_empty_part:                                              [ OK ]

All tests passed.
```

**Comment by vavrusa** [comment]
@alexey-milovidov can you take a look? This should fix autotests again.

**Pull request merged**